### PR TITLE
DOC: explain null hypotheses in ttest functions

### LINKED ISSUE #12073: Add note in documentation
https://github.com/scipy/scipy/blob/adc4f4f7bab120ccfab9383aba272954a0a12fb0/scipy/stats/stats.py#L4999

So in computing this test you are implicitly assuming that the null hipothesis of the test is build like:
```tex
 <mean1> - <mean2> = 0.
```
But one could build it the other way:
```tex
<mean2> - <mean1> = 0.
```

Now this supposes no issue if we are only interested on the two-sided test. However since it is possible to derive the result of a one-side test from the two-sided test result, it is important to know which of the assumptions you are making in order to avoid miss interpretations of the results.

So it is okay if you make the hipothesis like this
 ```tex
d = <mean1> - <mean2> = 0.
```
but that information should be available and clear in the documentation.
Thanks in advance

### PR DESCRIPTION
**Pull request by jacoblapenna**
Added sentences to documentation of various ttest_ methods in stats to better describe how the null hypotheses are formed in the source code.

#### Reference issue
Closes gh-12073

#### What does this implement/fix?

Removes ambiguity associated with how the null hypothesis is formed behind various t-test functions, particularly when calculating a one-sided statistic.

#### Additional information

Hope this helps! Let me know what I can do better, thanks!

### CODE REVIEW

File: scipy/stats/stats.py [hunk 1]
```diff
@@ -6673,7 +6692,7 @@ def chisquare(f_obs, f_exp=None, ddof=0, axis=0):
     and expected frequencies should be at least 5. If one or more frequencies
     are less than 5, Fisher's Exact Test can be used with greater statistical
     power. According to [3]_, the total number of samples is recommended to be
-    greater than 13, otherwise a table-based method of obtaining p-values is 
+    greater than 13, otherwise a table-based method of obtaining p-values is
```
Comment (tirthasheshpatel) [inline scipy/stats/stats.py:6695]: These docs have been changed in gh-14012. If you revert this change, I think the conflicts should go away:

```suggestion
    greater than 13, otherwise a table-based method of obtaining p-values is 
```

File: scipy/stats/stats.py [hunk 2]
```diff
@@ -6694,8 +6713,8 @@ def chisquare(f_obs, f_exp=None, ddof=0, axis=0):
            Statistics". Chapter 8.
            https://web.archive.org/web/20171022032306/http://vassarstats.net:80/textbook/ch8pt1.html
     .. [2] "Chi-squared test", https://en.wikipedia.org/wiki/Chi-squared_test
-    .. [3] Pearson, Karl. "On the criterion that a given system of deviations from the probable 
-           in the case of a correlated system of variables is such that it can be reasonably 
+    .. [3] Pearson, Karl. "On the criterion that a given system of deviations from the probable
+           in the case of a correlated system of variables is such that it can be reasonably
```
Comment (tirthasheshpatel) [inline scipy/stats/stats.py:6717]: Same here:

```suggestion
    .. [3] Pearson, Karl. "On the criterion that a given system of deviations from the probable 
           in the case of a correlated system of variables is such that it can be reasonably 
```

File: scipy/stats/stats.py [hunk 3]
```diff
@@ -5505,6 +5505,12 @@ def ttest_1samp(a, popmean, axis=0, nan_policy='propagate',
     pvalue : float or array
         Two-sided p-value.
 
+    Notes
+    -----
+    The null hypothesis is determined as ``numpy.mean(a, axis) - popmean = 0``.
```
Comment (tirthasheshpatel) [inline scipy/stats/stats.py:5510]: These look good. One good (but not necessary) change would be use `np` instead of the full namespace `numpy` as `numpy` is imported as `np` everywhere in SciPy:

```suggestion
    The null hypothesis is determined as ``np.mean(a, axis) - popmean = 0``.
```

**Review by tirthasheshpatel — COMMENTED** [review]
Hi, @jacoblapenna, thanks for contributing! I have been working on gh-13958 so I thought I would help you with this. The docs look good to me. I have made a few comments to resolve the conflict and a small non-blocking suggestion.

**Comment by SRDdev** [comment]
Hello @tirthasheshpatel,
Is this issue currently assigned to anyone , if not can I work on this issue?
I am currently searching for a "good-first-issue" to start open source journey.

**Comment by tirthasheshpatel** [comment]
This hasn't been assigned to anyone to my knowledge. Feel free to take this up and complete it.

**Review by tirthasheshpatel — APPROVED** [review]
@mdhaber Thanks for resolving the conflict and resuming work on this. My only comment was changing `numpy` -> `np` (just a nit). Otherwise, the docs look good to me. If you are happy with it too, feel free to merge.

**Comment by mdhaber** [comment]
Since this was written, the alternative hypotheses have been clearly defined. I think all we need to do here is state the direction in which the statistic is calculated.

File: scipy/stats/_stats_py.py [hunk 4]
```diff
@@ -5761,6 +5761,12 @@ def ttest_1samp(a, popmean, axis=0, nan_policy='propagate',
     pvalue : float or array
         Two-sided p-value.
 
+    Notes
+    -----
+    The null hypothesis is determined as ``numpy.mean(a, axis) - popmean = 0``.
+    This is significant when determining which side of the t-distribution
+    the one-sided statistic should be taken from (``less`` or ``greater``).
```
Comment (mdhaber) [inline scipy/stats/_stats_py.py:5768]:
```suggestion
    The statistic is calculated as ``(np.mean(a) - popmean)/se``, where
    ``se`` is the standard error. Therefore, the statistic will be positive
    when the sample mean is greater than the population mean and negative when
    the population mean is greater than the sample mean.
```

File: scipy/stats/_stats_py.py [hunk 5]
```diff
@@ -5931,7 +5937,9 @@ def ttest_ind_from_stats(mean1, std1, nobs1, mean2, std2, nobs2,
 
     Notes
     -----
-    .. versionadded:: 0.16.0
+    The null hypothesis is determined as ``mean1 - mean2 = 0``. This is
+    significant when determining which side of the t-distribution
+    the one-sided statistic should be taken from (``less`` or ``greater``).
```
Comment (mdhaber) [inline scipy/stats/_stats_py.py:5942]:
```suggestion
    The statistic is calculated as ``(mean1 - mean2)/se``, where ``se`` is the 
    standard error. Therefore, the statistic will be positive when `mean1` is 
    greater than `mean2` and negative when `mean2` is greater than `mean1`.
```

File: scipy/stats/_stats_py.py [hunk 6]
```diff
@@ -6190,6 +6198,11 @@ def ttest_ind(a, b, axis=0, equal_var=True, nan_policy='propagate',
     recommended if the underlying distribution is long-tailed or contaminated
     with outliers [4]_.
 
+    The null hypothesis is determined as
+    ``numpy.mean(a, axis) - numpy.mean(b, axis) = 0``. This is significant when
+    determining which side of the t-distribution the one-sided statistic
+    should be taken from (``less`` or ``greater``).
+
```
Comment (mdhaber) [inline scipy/stats/_stats_py.py:6205]:
```suggestion
    The statistic is calculated as ``(np.mean(a) - np.mean(b))/se``, where 
    ``se`` is the standard error. Therefore, the statistic will be positive
    when the sample mean of `a` is greater than the sample mean of `b` and 
    negative when the sample mean of `b` is greater than the sample mean of 
    `a`.
```

File: scipy/stats/_stats_py.py [hunk 7]
```diff
@@ -6579,6 +6592,10 @@ def ttest_rel(a, b, axis=0, nan_policy='propagate', alternative="two-sided"):
     hypothesis of equal averages. Small p-values are associated with
     large t-statistics.
 
+    The null hypothesis is determined as ``numpy.mean(a - b, axis) = 0``.
+    This is significant when determining which side of the t-distribution
+    the one-sided statistic should be taken from (``less`` or ``greater``).
+
```
Comment (mdhaber) [inline scipy/stats/_stats_py.py:6598]:
```suggestion
    The statistic is calculated as ``np.mean(a - b)/se``, where ``se`` is the
    standard error. Therefore, the statistic will be positive when the sample
    mean of ``a - b`` is positive and negative when the sample mean of
    ``a - b`` is negative.

```

**Review by mdhaber — APPROVED** [review]
The user already needs to assume that statements about samples really refer to each axis-slice of the sample, so I removed `axis` from the statements to keep it simple. Do these still look OK to you @tirthasheshpatel?

**Comment by tirthasheshpatel** [comment]
> Do these still look OK to you

Looks good. I have committed them. I guess this is ready to go once the tests pass.

File: scipy/stats/_stats_py.py [hunk 8]
```diff
@@ -5931,7 +5938,9 @@ def ttest_ind_from_stats(mean1, std1, nobs1, mean2, std2, nobs2,
 
     Notes
     -----
-    .. versionadded:: 0.16.0
+    The statistic is calculated as ``(mean1 - mean2)/se``, where ``se`` is the 
+    standard error. Therefore, the statistic will be positive when `mean1` is
```
Comment (tirthasheshpatel) [inline scipy/stats/_stats_py.py:5942]:
```suggestion
    The statistic is calculated as ``(mean1 - mean2)/se``, where ``se`` is the
    standard error. Therefore, the statistic will be positive when `mean1` is
```

File: scipy/stats/_stats_py.py [hunk 9]
```diff
@@ -6190,6 +6199,11 @@ def ttest_ind(a, b, axis=0, equal_var=True, nan_policy='propagate',
     recommended if the underlying distribution is long-tailed or contaminated
     with outliers [4]_.
 
+    The statistic is calculated as ``(np.mean(a) - np.mean(b))/se``, where
```
Comment (tirthasheshpatel) [inline scipy/stats/_stats_py.py:6202]:
```suggestion
    The statistic is calculated as ``(np.mean(a) - np.mean(b))/se``, where
```

File: scipy/stats/_stats_py.py [hunk 10]
```diff
@@ -6190,6 +6199,11 @@ def ttest_ind(a, b, axis=0, equal_var=True, nan_policy='propagate',
     recommended if the underlying distribution is long-tailed or contaminated
     with outliers [4]_.
 
+    The statistic is calculated as ``(np.mean(a) - np.mean(b))/se``, where 
+    ``se`` is the standard error. Therefore, the statistic will be positive
+    when the sample mean of `a` is greater than the sample mean of `b` and 
+    negative when the sample mean of `a` is less than the sample mean of
```
Comment (tirthasheshpatel) [inline scipy/stats/_stats_py.py:6205]:
```suggestion
    when the sample mean of `a` is greater than the sample mean of `b` and
    negative when the sample mean of `a` is less than the sample mean of
```

File: scipy/stats/_stats_py.py [hunk 11]
```diff
@@ -6190,6 +6199,11 @@ def ttest_ind(a, b, axis=0, equal_var=True, nan_policy='propagate',
     recommended if the underlying distribution is long-tailed or contaminated
     with outliers [4]_.
 
+    The statistic is calculated as ``(np.mean(a) - np.mean(b))/se``, where
+    ``se`` is the standard error. Therefore, the statistic will be positive
+    when the sample mean of `a` is greater than the sample mean of `b` and
+    negative when the sample mean of `a` is less than the sample mean of
+    `b`.
```
Comment (mdhaber) [inline scipy/stats/_stats_py.py:6206]: I think this will fix:
```/home/circleci/repo/build/testenv/lib/python3.8/site-packages/scipy/stats/_stats_py.py:docstring of scipy.stats._stats_py.ttest_ind:19: WARNING: citation not found: R3566833beaa2-1```

```suggestion
    `b`.

```

**Comment by mdhaber** [comment]
CI is green now. Thanks @jacoblapenna @tirthasheshpatel!
